kiwi/storage/subformat/vagrant_base.py

```diff
@@ -115,29 +124,32 @@ def create_image_format(self):
         * creation of box Vagrantfile
         * creation of result format tarball from the files created above
         """
-        self.temp_image_dir = mkdtemp(prefix='kiwi_vagrant_box.')
-
-        box_img_files = self.create_box_img(self.temp_image_dir)
-
-        metadata_json = os.sep.join([self.temp_image_dir, 'metadata.json'])
-        with open(metadata_json, 'w') as meta:
-            meta.write(self._create_box_metadata())
-
-        vagrantfile = os.sep.join([self.temp_image_dir, 'Vagrantfile'])
-        with open(vagrantfile, 'w') as vagrant:
-            vagrant.write(self._create_box_vagrantconfig())
-
-        Command.run(
-            [
-                'tar', '-C', self.temp_image_dir,
-                '-czf', self.get_target_file_path_for_format(self.image_format),
-                os.path.basename(metadata_json),
-                os.path.basename(vagrantfile)
-            ] + [
-                os.path.basename(box_img_file)
-                for box_img_file in box_img_files
-            ]
-        )
+        if self.image_format and self.provider:
```

Comment:
[suggestion] The function will blow up if called from a child class that did not properly set `image_format` or `provider`. With your change it will silently do nothing. I would advise not to do that.

Reply:
yes you are fully correct, I'll adapt

Reply:
done, raising on improper setup